src/main.c

```diff
@@ -317,9 +322,13 @@ int main(int argc, char *argv[])
 	printf("\n");
 
 	// set up system state
-	// 512K of RAM
+	// RAM sizes come from config, default 2 Meg for each kind of memory
 	int i;
-	if ((i = state_init(2048*1024, 2048*1024)) != STATE_E_OK) {
+	int base_memory = fbc_get_int("memory", "base_memory");
```

Comment:
It probably doesn't make sense to allow `base_memory` to be less than 512K. That was the lowest-memory 3B1 available (actually that was the original 7300). Most machines would have had 1MB or 2MB.
I don't know how the RAM count will behave with 1K increments either... exercise for the reader? :grin: